src/components/decision-engine/index.js

```diff
@@ -35,14 +38,42 @@ class DecisionEngine {
   }
 
   _sendBlocks (env, cb) {
+    // split into messges of max 512 * 1024 bytes
```

Comment:
Ah yes, because of https://github.com/ipfs/go-ipfs/blob/master/exchange/bitswap/network/ipfs_impl.go#L186

We have a limit to the maximum message size you can send.